src/NodeVisitor/MacroAutoImportNodeVisitor.php

```diff
@@ -28,41 +21,18 @@
  */
 final class MacroAutoImportNodeVisitor implements NodeVisitorInterface
 {
-    private $inAModule = false;
-    private $hasMacroCalls = false;
+    public function __construct()
+    {
+        trigger_deprecation('twig/twig', '3.15', 'The "%s" class is deprecated.', __CLASS__);
+    }
 
     public function enterNode(Node $node, Environment $env): Node
     {
-        if ($node instanceof ModuleNode) {
-            $this->inAModule = true;
-            $this->hasMacroCalls = false;
-        }
-
         return $node;
     }
 
     public function leaveNode(Node $node, Environment $env): Node
     {
-        if ($node instanceof ModuleNode) {
-            $this->inAModule = false;
-            if ($this->hasMacroCalls) {
-                $node->getNode('constructor_end')->setNode('_auto_macro_import', new ImportNode(new NameExpression('_self', 0), new AssignNameExpression('_self', 0), 0));
```

Comment:
Where is this going ?

Reply:
It's here: https://github.com/twigphp/Twig/pull/4402/files#diff-ad87623d49894167ed5e4ecd1814164bdbe5c4c250e6b2ad9da415edd22e8e85R965

Reply:
but this does not add anymore the auto-import at the end of the constructor AFAICT.

Reply:
Because it's not needed anymore